src/components/GameTile.vue

```diff
@@ -0,0 +1,354 @@
+<script setup>
+import { reactive, defineProps, computed } from "vue";
+import CustomButton from "@/components/CustomButton.vue";
+
+const props = defineProps({
+  tile: {
+    type: Object,
+    required: true,
+  },
+  currency: {
+    type: String,
+    default: "USD",
+    required: true,
+  },
+});
+
+const cartIcon = {
+  initial: require(`@/assets/icon/icon_shopping-cart.svg`),
+  added: require(`@/assets/icon/icon_add-shopping-cart.svg`),
+};
+
+const addToCartButton = reactive({
+  iconSource: cartIcon.initial,
+  iconAlternativeText: "Icon that shows a shopping cart.",
+  label: "ADD TO CART",
+  disabled: false,
+});
+
+const platformIconStyle = computed(() => {
+  if (props.tile?.platform) {
+    try {
+      const iconUrl = require(`@/assets/logo/logo_${props.tile.platform}.svg`);
+      return { "--platform-icon": `url(${iconUrl})` };
+    } catch (e) {
+      console.error(`Cannot load icon for platform: ${props.tile.platform}`);
+      return {};
+    }
+  }
+  return {};
+});
+
+const hasDiscount = computed(
+  () => (props?.tile?.discountInPercentage ?? 0) > 0
+);
+
+const discountedPrice = computed(() => {
+  if (!hasDiscount.value) return props.tile?.price.toFixed(2);
+
+  const discount = Math.max(Math.min(props.tile?.discountInPercentage, 100), 1);
+  const finalPrice = props.tile?.price - (props.tile?.price * discount) / 100;
+
+  return Math.max(finalPrice.toFixed(2), 0.01).toFixed(2);
+});
+
+function truncateDescription(text, maxLength) {
+  if (text.length <= maxLength) return text;
+  let truncated = text.slice(0, maxLength);
+  const lastSpace = truncated.lastIndexOf(" ");
+  if (lastSpace > 0) {
+    truncated = truncated.slice(0, lastSpace);
+  }
+  return truncated + "...";
+}
+
+function onAddToCartButtonClick() {
+  addToCartButton.disabled = true;
+  addToCartButton.iconSource = cartIcon.added;
+  addToCartButton.iconAlternativeText = "Icon that shows an added to cart ";
+  addToCartButton.label = "ADDED";
+}
+
+function handleNavigateToGamePage() {
+  console.log("Navigate to game page");
+}
+</script>
+
+<template>
+  <div
+    :style="platformIconStyle"
+    :class="$style['game-tile']"
+    @click="handleNavigateToGamePage"
+  >
+    <div :class="$style['game-tile__image-container']">
+      <img
+        :class="$style['game-tile__image']"
+        :src="require(`@/assets/${tile.imagePath}`)"
+        :alt="`Cover art for ${tile.title}`"
+      />
+    </div>
+    <div :class="[$style.content, $style['game-tile__content']]">
+      <div :class="$style['content__title']">{{ tile.title }}</div>
+      <div :class="$style['content__genres']">
+        <template v-for="genre in tile.genres" :key="genre.name">
```

Comment:
So genre could have id as well... it is more safe then
```suggestion
        <template v-for="genre in tile.genres" :key="genre.id">
```